lib/graphql/query/executor.rb

```diff
@@ -31,7 +33,11 @@ def execute
         execution_strategy = execution_strategy_class.new
 
         query.context.execution_strategy = execution_strategy
-        data_result = execution_strategy.execute(operation, root_type, query)
+        data_result = begin
+          execution_strategy.execute(operation, root_type, query)
+        rescue PropagateNull
```

Comment:
This was moved outside of the execution strategy because at this point all the promises have been resolved in graphql-batch.